test/src/test/java/org/corfudb/integration/CorfuReplicationUpgradeIT.java

```diff
@@ -937,4 +938,119 @@ private void verifyVersion(CorfuStore corfuStore, String expectedVersion, boolea
+                LogReplicationMetadataManager.REPLICATION_STATUS_TABLE,
+                LogReplicationMetadata.ReplicationStatusKey.class,
+                LogReplicationMetadata.ReplicationStatusVal.class,
+                null,
+                TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+
+        CountDownLatch statusUpdateLatch = new CountDownLatch(2);
+        ReplicationStatusListener standbyListener = new ReplicationStatusListener(statusUpdateLatch, false);
+        corfuStoreStandby.subscribeListener(standbyListener, LogReplicationMetadataManager.NAMESPACE,
+                LogReplicationMetadataManager.LR_STATUS_STREAM_TAG);
+
+        setupVersionTable(corfuStoreActive, false);
+        setupVersionTable(corfuStoreStandby, false);
+
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_ACTIVE;
+        startActiveLogReplicator();
+
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_STANDBY;
+        startStandbyLogReplicator();
+
+        log.info(">> Wait ... Snapshot log replication in progress ...");
+        verifyDataOnStandby(NUM_WRITES);
+
+        // Verify that snapshot sync was triggered by checking the number of
+        // updates to the ReplicationStatus table on the standby.
+        latchSnapshotSyncPlugin.await();
+        validateSnapshotSyncPlugin(snapshotSyncPluginListener);
+        statusUpdateLatch.await();
+
+        Assert.assertEquals(2, standbyListener.getAccumulatedStatus().size());
+        Assert.assertTrue(standbyListener.getAccumulatedStatus().get(1));
+        Assert.assertFalse(standbyListener.getAccumulatedStatus().get(0));
+        corfuStoreStandby.unsubscribeListener(standbyListener);
+        corfuStoreStandby.unsubscribeListener(snapshotSyncPluginListener);
+
+        // Upgrade the standby site
+        log.info(">> Upgrading the standby site ...");
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_STANDBY;
+        upgradeSite(false, corfuStoreStandby);
+        verifyVersion(corfuStoreStandby, UPGRADE_VERSION_STRING, true);
+        verifyVersion(corfuStoreActive, VERSION_STRING, false);
+
+        latchSnapshotSyncPlugin = new CountDownLatch(2);
+        snapshotSyncPluginListener = new SnapshotSyncPluginListener(latchSnapshotSyncPlugin);
+        subscribeToSnapshotSyncPluginTable(snapshotSyncPluginListener);
+
+        statusUpdateLatch = new CountDownLatch(2);
+        standbyListener = new ReplicationStatusListener(statusUpdateLatch, false);
+        corfuStoreStandby.subscribeListener(standbyListener, LogReplicationMetadataManager.NAMESPACE,
+                LogReplicationMetadataManager.LR_STATUS_STREAM_TAG);
+
+        // Trigger a snapshot sync by stopping the active LR and running a CP+trim
+        stopActiveLogReplicator();
+        checkpointAndTrim(true);
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_ACTIVE;
+        startActiveLogReplicator();
+
+        // Verify that snapshot sync between the different versions was
```

Comment:
What two different versions? here both are the same version..

Reply:
will work on a better testcase.